src/renderer/coremods/badges/badge.tsx

```diff
@@ -71,108 +64,137 @@ export const getBadgeComponent = async (): Promise<(args: BadgeProps) => React.R
   }
   const { profileBadge22 } = badgeClassMod;
 
-  return ({ color, tooltip, tooltipPosition, className, children, gap, onClick }: BadgeProps) => {
-    return (
-      <Clickable className={clickableClass.clickable} onClick={onClick || (() => undefined)}>
+  const child = (
+    <div
+      className={`${profileBadge22} replugged-badge ${className || ""}`}
+      style={{ color: `#${color || "7289da"}` }}>
+      {children}
+    </div>
+  );
+  return (
+    <Clickable className={clickableClass.clickable} onClick={onClick || (() => undefined)}>
+      {tooltip ? (
         <Tooltip
           text={tooltip}
           position={tooltipPosition || "top"}
           spacing={gap === false ? 0 : 12}>
-          {(props) => (
-            <span {...props}>
-              <div
-                className={`${profileBadge22} replugged-badge ${className || ""}`}
-                style={{ color: `#${color || "7289da"}` }}>
-                {children}
-              </div>
-            </span>
-          )}
+          {child}
         </Tooltip>
-      </Clickable>
-    );
-  };
+      ) : (
+        child
+      )}
```

Comment:
There is a `shouldShow` prop that does this for you lol

```jsx
<Tooltip
  shouldShow={Boolean(tooltip)}
  text={tooltip!}
  position={tooltipPosition || "top"}
  spacing={gap === false ? 0 : 12}>
  {child}
</Tooltip>
```

Reply:
I think the current way of doing it is better. No reason to include extra elements if they're not being used.

Reply:
It's not an extra element... it's just a prop that's already added. You';d just have to replace those lines with the codeblock I sent

Reply:
If the element isn't doing anything then it is an extra element.

Reply:
You're literally reducing lines into one prop line?